src/Container/PdoSnapshotStoreFactory.php

```diff
@@ -59,9 +59,25 @@ public static function __callStatic(string $name, array $arguments): PdoSnapshot
     public function __invoke(ContainerInterface $container): PdoSnapshotStore
     {
         $config = $container->get('config');
+
+        /** repair legacy connection_service config key*/
+        (function (&$config): void {
+            foreach ($this->dimensions() as $dimension) {
+                $config = &$config[$dimension];
+            }
+
+            $config = &$config[$this->configId] ?? [];
+
+            if (! isset($config['connection']) && isset($config['connection_service'])) {
+                $config['connection'] = $config['connection_service'];
+            }
+
+            unset($config['connection_service']);
+        })($config);
```

Comment:
I'm not sure if we should unset the `connection_service` key. Maybe someone is referencing it.

Reply:
Agree, @basz can you remove this unset call?